projects/plugins/boost/app/modules/Modules_Setup.php

```diff
@@ -103,6 +113,10 @@ public function init_modules() {
 
 			Setup::add( $module->feature );
 
+			if ( $module->has_sub_modules() ) {
+				$module->feature->setup_sub_modules();
+			}
```

Comment:
Should the parent module define the setup or should that be somewhere else? I think it should be outside, as this would be duplicate code for every module that has sub modules. However, I'm not sure where it should be.

Maybe `Has_Sub_Modules` should only have a `get_sub_modules_state` and `get_sub_modules` methods. Then here we'd get modules and run their setup?